fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -1305,13 +1306,20 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 			    .detail("KnobRocksDBWriteRateLimiterAutoTune", SERVER_KNOBS->ROCKSDB_WRITE_RATE_LIMITER_AUTO_TUNE)
 			    .detail("ColumnFamily", cf->GetName());
 			if (g_network->isSimulated()) {
-				// The current thread and main thread are same when the code runs in simulation.
-				// blockUntilReady() is getting the thread into deadlock state, so directly calling
-				// the metricsLogger.
-				a.metrics =
-				    rocksDBMetricLogger(
-				        id, sharedState, options.statistics, perfContextMetrics, db, readIterPool, &a.counters, cf) &&
-				    flowLockLogger(id, a.readLock, a.fetchLock) && refreshReadIteratorPool(readIterPool);
+				if (SERVER_KNOBS->ROCKSDB_METRICS_IN_SIMULATION) {
+					// The current thread and main thread are same when the code runs in simulation.
+					// blockUntilReady() is getting the thread into deadlock state, so directly calling
+					// the metricsLogger.
+					a.metrics = rocksDBMetricLogger(id,
+					                                sharedState,
+					                                options.statistics,
+					                                perfContextMetrics,
+					                                db,
+					                                readIterPool,
+					                                &a.counters,
+					                                cf) &&
+					            flowLockLogger(id, a.readLock, a.fetchLock) && refreshReadIteratorPool(readIterPool);
```

Comment:
a.metrics is holding three actors - rocksDBMetricLogger, flowLockLogger, refreshReadIteratorPool.
You want to block only rocksDBMetricLogger based on knob and not the remaining. 
This is a messy code, I am working on a diff to rearrange that.

Reply:
Good idea! Updated.